snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedDescriptionAcceptabilityUpdateRequest.java

```diff
@@ -59,49 +63,95 @@ protected boolean canUpdate(TransactionContext context) {
 				.filterByRefSetType(Collections.singleton(SnomedRefSetType.LANGUAGE))
+				.sortBy(SnomedRefSetMemberIndexEntry.Fields.ID)
 				.build()
-				.execute(context);
+				.execute(context)
+				.getItems();
 		
-		final Map<String, Acceptability> newLanguageMembersToCreate = newHashMap(newAcceptabilityMap);
+		final Map<String, Acceptability> newLanguageMembersToCreate = newHashMap();
 		final ModuleIdProvider moduleIdSupplier = context.service(ModuleIdProvider.class);
+		final String moduleId = moduleIdSupplier.apply(componentToUpdate);
 
 		// check if there are existing matches
-		for (SnomedReferenceSetMember existingMember : existingMembers) {
-			final String acceptabilityId = (String) existingMember.getProperties().get(SnomedRf2Headers.FIELD_ACCEPTABILITY_ID);
-			final Acceptability acceptability = Acceptability.getByConceptId(acceptabilityId);
-			final String languageReferenceSetId = existingMember.getReferenceSetId();
+		for (String languageReferenceSet : newAcceptabilityMap.keySet()) {
 			
-			if (null == acceptability) {
-				continue;	
-			}
+			final Acceptability newAcceptability = newAcceptabilityMap.get(languageReferenceSet);
+			final List<SnomedReferenceSetMember> refsetMatches = existingMembers.stream()
+					.filter(m -> Objects.equal(languageReferenceSet, m.getReferenceSetId()))
+					.collect(Collectors.toList());
 			
-			final SnomedRefSetMemberIndexEntry.Builder updatedMember = SnomedRefSetMemberIndexEntry.builder(existingMember);
-			final SnomedRefSetMemberIndexEntry oldRevision = updatedMember.build();
+			SnomedReferenceSetMember existingMember = null;
 			
-			if (acceptability.equals(newLanguageMembersToCreate.get(languageReferenceSetId))) {
-				// Exact match: make sure that the member is active
-				if (ensureMemberActive(context, existingMember, updatedMember)) {
+			if (!refsetMatches.isEmpty()) {
+				// Case 1: Match(es) exist for language reference set
+				final List<SnomedReferenceSetMember> acceptabilityMatches = refsetMatches.stream()
+						.filter(m -> Objects.equal(newAcceptability.getConceptId(), m.getProperties().get(SnomedRf2Headers.FIELD_ACCEPTABILITY_ID)))
+						.collect(Collectors.toList());
+				
+				if (acceptabilityMatches.isEmpty()) {
+					// Case 1.1: No match for acceptability; use an existing member preferably an extension one
+					existingMember = select(refsetMatches, moduleId, Concepts.MODULE_SCT_CORE);
+					final SnomedRefSetMemberIndexEntry.Builder updatedMember = SnomedRefSetMemberIndexEntry.builder(existingMember);
+					final SnomedRefSetMemberIndexEntry oldRevision = updatedMember.build();
+					
+					// Change acceptability, set status to active if required, place it in the supplied module
+					updatedMember.field(SnomedRf2Headers.FIELD_ACCEPTABILITY_ID, newAcceptability.getConceptId());
+					ensureMemberActive(context, existingMember, updatedMember);
+					updateModule(context, existingMember, updatedMember, moduleIdSupplier.apply(componentToUpdate));
```

Comment:
As `moduleIdSupplier.apply(componentToUpdate)` is already extracted to a local variable above, we can make this line shorter:

```suggestion
					updateModule(context, existingMember, updatedMember, moduleId);
```